File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private ArrayList<Integer> arr = new ArrayList<Integer>();

    public int deepestLeavesSum(TreeNode root) {
        sum(root, 0);
        return arr.get(arr.size() - 1);
    }

    private void sum(TreeNode root, int level) {
        if (level >= arr.size()) {
            arr.add(root.val);
        } else {
            arr.set(level, arr.get(level) + root.val);
        }

        if (root.left != null)
            sum(root.left, level + 1);
        if (root.right != null)
            sum(root.right, level + 1);
    }

}

File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    private val arr = mutableListOf<Int>()
    fun deepestLeavesSum(root: TreeNode?): Int {
        if (root == null) return 0
        sum(root, 0)
        return arr.last()
    }

    private fun sum(root: TreeNode, level: Int) {
        if (level >= arr.size) {
            arr.add(root.`val`)
        } else if (level == arr.lastIndex) {
            arr[level] = arr[level] + root.`val`
        }
        root.left?.let { sum(it, level + 1) }
        root.right?.let { sum(it, level + 1) }
    }
}